File: 05- May/02- Sign of the Product of an Array/02- Sign of the Product of an Array (Noura Algohary).py
# Author: Noura Algohary
class Solution:
    def arraySign(self, nums: List[int]) -> int:
        signCounter = 0

        for num in nums:
            # if one zero appears, the result is zero
            if num == 0:
                return 0
            elif num < 0:
                signCounter += 1
        
        # if negative numbers are of odd count, the result is negative
        return 1 if signCounter % 2 == 0 else -1

File: 05- May/02- Sign of the Product of an Array/02- Sign of the Product of an Array (Noura Algohary).cpp
// Author: Noura Algohary
class Solution {
public:
    int arraySign(vector<int>& nums) {
        int signCounter = 0;

        for(int num : nums)
        {
            // if one zero appears, the result is zero
            if(num==0)
                return 0;
            else if(num<0)
                signCounter++;
        }

        // if negative numbers are of odd count, the result is negative
        if (signCounter%2==0)
            return 1;
        else
            return -1;
    }
};